pycomlink/processing/wet_dry/nearby_wetdry.py

```diff
@@ -0,0 +1,216 @@
+import xarray as xr
+import numpy as np
+from tqdm import tqdm
+import pycomlink.spatial.helper as spatial
+
+
+def calc_distance_between_cml_endpoints(
+    cml_ids,
+    site_a_latitude,
+    site_a_longitude,
+    site_b_latitude,
+    site_b_longitude,
+):
+    """
+    calculating the distance between all start and endpoints of a network
+    of CMLs using the Haversine distance forumla
+    ----------
+    cml_ids : list of str or int
+         ids of CMLs
+    site_a_latitude : list or array
+        latitude values of site a
+    site_a_longitude : list or array
+        longitude values of site a
+    site_b_latitude : list or array
+        latitude values of site b
+    site_b_longitude : list or array
+        longitude values of site b
+    Returns
+    -------
+    xarray.Dataset
+        distances between all start and endpoints in four variables, a_to_all_a,
+        a_to_all_b, b_to_all_a, b_to_all_b
+    """
+
+    n_cmls = len(cml_ids)
+    ds = xr.Dataset(
+        data_vars=dict(
+            a_to_all_a=(["cml_id1", "cml_id2"], np.full([n_cmls, n_cmls], np.nan)),
+            a_to_all_b=(["cml_id1", "cml_id2"], np.full([n_cmls, n_cmls], np.nan)),
+            b_to_all_a=(["cml_id1", "cml_id2"], np.full([n_cmls, n_cmls], np.nan)),
+            b_to_all_b=(["cml_id1", "cml_id2"], np.full([n_cmls, n_cmls], np.nan)),
+        ),
+        coords=dict(cml_id1=cml_ids, cml_id2=cml_ids),
+    )
+    for i, cmlid in tqdm(enumerate(cml_ids)):
+        ds.loc[dict(cml_id1=cmlid)]["a_to_all_a"][:] = spatial.haversine(
+            np.array(site_a_longitude[i]),
+            np.array(site_a_latitude[i]),
+            np.array(site_a_longitude),
+            np.array(site_a_latitude),
+        )
+        ds.loc[dict(cml_id1=cmlid)]["a_to_all_b"][:] = spatial.haversine(
+            np.array(site_a_longitude[i]),
+            np.array(site_a_latitude[i]),
+            np.array(site_b_longitude),
+            np.array(site_b_latitude),
+        )
+        ds.loc[dict(cml_id1=cmlid)]["b_to_all_a"][:] = spatial.haversine(
+            np.array(site_b_longitude[i]),
+            np.array(site_b_latitude[i]),
+            np.array(site_a_longitude),
+            np.array(site_a_latitude),
+        )
+        ds.loc[dict(cml_id1=cmlid)]["b_to_all_b"][:] = spatial.haversine(
+            np.array(site_b_longitude[i]),
+            np.array(site_b_latitude[i]),
+            np.array(site_b_longitude),
+            np.array(site_b_latitude),
+        )
+    # set distance between endpoints of one cml to 0km to not be rejected when
+    # radius r for the nearby approach is larger than r
+    ds = xr.where(
```

Comment:
From the comment above I do not fully understand what this does. Does it put zeros along the diagonal of the 2D distance matrix?

Is this required because we want zero also for distance between site_a and site_b of the same CML, right?

Maybe you can make this clearer in the comment here and also in the docstring of the function because, even though it is intuitive that the diagonal is set to zero, it is not explicitly clear that this is done and why.

Reply:
Yes, your assumption is true.  As result, the distance between site_a and site_b of the same CML is set to 0 km.
I have updated the docstring and the comment in the before the actual command doing this.